File: army/Attack/AttackRogue.cpp
#include "AttackRogue.h"

AttackRogue::AttackRogue() {}

AttackRogue::~AttackRogue() {}

void AttackRogue::attack(Unit* attacker, Unit* enemy) {
    enemy->takePhysicalDamage(attacker->getDamage());
}

void AttackRogue::counterAttack(Unit* attacker, Unit* enemy) {
    enemy->takePhysicalDamage(attacker->getDamage() / 2);
}

File: army/JavaArmy/src/com/maksklaban/armymax/attack/AttackRogue.java
package com.maksklaban.armymax.attack;

import com.maksklaban.armymax.exceptions.*;
import com.maksklaban.armymax.state.*;
import com.maksklaban.armymax.units.*;

public class AttackRogue extends NormalAttack {
    public AttackRogue() {}

    public void attack(Unit attacker, Unit enemy) throws ArmyException {
        enemy.takePhysicalDamage(attacker.getDamage());
    }

    public void counterAttack(Unit attacker, Unit enemy) throws ArmyException {
        enemy.takePhysicalDamage(attacker.getDamage() / 2);
    }
}
